services/transactionpool/test/get_transactions_for_ordering_test.go

```diff
@@ -107,20 +108,37 @@ func TestGetTransactionsForOrderingDoesNotWaitForAdditionalTransactionsIfContext
 	})
 }
 
-func TestGetTransactionsForOrderingOnGenesisBlockReturnsZeroTransactions(t *testing.T) {
+func TestGetTransactionsForOrdering_FiltersOutTooBigProtocolVersion(t *testing.T) {
 	with.Concurrency(t, func(ctx context.Context, parent *with.ConcurrencyHarness) {
 		h := newHarness(parent).start(ctx)
-		h.handleForwardFrom(ctx, otherNodeKeyPair, builders.TransferTransaction().Build())
+		h.handleForwardFrom(ctx, otherNodeKeyPair, builders.TransferTransaction().WithProtocolVersion(config.MAXIMAL_PROTOCOL_VERSION_SUPPORTED_VALUE+5).Build())
 
 		out, err := h.txpool.GetTransactionsForOrdering(ctx, &services.GetTransactionsForOrderingInput{
-			CurrentBlockHeight:      1,
+			BlockProtocolVersion:    config.MAXIMAL_PROTOCOL_VERSION_SUPPORTED_VALUE,
+			CurrentBlockHeight:      2,
 			PrevBlockTimestamp:      0,
 			MaxNumberOfTransactions: 1,
 		})
 
 		require.NoError(t, err, "GetTransactionsForOrdering should not fail")
 		require.Zero(t, len(out.SignedTransactions), "number of transactions should be zero")
-		require.NotZero(t, out.ProposedBlockTimestamp, "proposed block timestamp should not be zero")
+	})
+}
+
+func TestGetTransactionsForOrdering_SucceedsForSmallerProtocolVersion(t *testing.T) {
```

Comment:
the intent of this test is not very clear to me. I think for clarity would be better to define a local variable called `currentProtocolVersion` and initialize it arbitrarily. the use of MAX inline is confusing imho.

second, the name is cryptic - maybe instead of "succeeds" say "ReturnsEarlierProtocolVersionTransactions" 

third, will this test work when MIN and MAX are the same version? maybe we don't need to support but in this case I'd add an assert at the top to clarify that you make this assumption.

lastly, if would be clearer maybe if you `-1` from the version number when creating the transaction and then passing the maximal version in `GetTransactionsForOrderingInput`. not critical..

Reply:
tx pool doesn't check minimal version. talking with Oded. Changed names.